File: testdata/php/simple.php
<?php

namespace App\Models;

use App\Contracts\Greeter;
use App\Services\Logger;

interface GreeterInterface
{
    public function greet(string $name): string;
}

trait Loggable
{
    public function log(string $message): void
    {
        echo $message;
    }
}

class SimpleGreeter implements GreeterInterface
{
    use Loggable;

    public function greet(string $name): string
    {
        $this->log("Greeting $name");
        return "Hello, $name!";
    }

    private function formatName(string $name): string
    {
        return ucfirst($name);
    }
}

enum Priority
{
    case Low;
    case Medium;
    case High;
}

function standalone(string $input): string
{
    $greeter = new SimpleGreeter();
    return $greeter->greet($input);
}

File: testdata/scala/Simple.scala
import scala.collection.mutable.ListBuffer

trait Greeter {
  def greet(name: String): String
}

class SimpleGreeter extends Greeter {
  override def greet(name: String): String = {
    s"Hello, $name!"
  }

  private def formatName(name: String): String = {
    name.capitalize
  }
}

object AppConfig {
  val defaultGreeting: String = "Hello"

  def createGreeter(): Greeter = {
    new SimpleGreeter()
  }
}

def main(args: Array[String]): Unit = {
  val greeter = AppConfig.createGreeter()
  println(greeter.greet("World"))
}
